src/interfaces/IReservoirRouter.sol

```diff
@@ -0,0 +1,113 @@
+pragma solidity 0.8.13;
+
+struct ExtraData {
+    uint64 token0PrecisionMultiplier;
+    uint64 token1PrecisionMultiplier;
+    uint64 amplificationCoefficient;
+}
+
+interface IReservoirRouter {
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                LIQUIDITY METHODS
+    //////////////////////////////////////////////////////////////////////////*/
+
+    function addLiquidity(
+        address tokenA,
+        address tokenB,
+        uint256 curveId,
+        uint amountADesired,
+        uint amountBDesired,
+        uint amountAMin,
+        uint amountBMin,
+        address to
+    ) external payable returns (uint256 amountA, uint256 amountB, uint256 liquidity);
+
+    function removeLiquidity(
+        address tokenA,
+        address tokenB,
+        uint256 curveId,
+        uint256 liquidity,
+        uint256 amountAMin,
+        uint256 amountBMin,
+        address to
+    ) external returns (uint256 amountA, uint256 amountB);
+
+    // the implementation these two functions take a lower priority
+    // will implement when we have the time, and does not block merge
+    // function addLiquiditySingle(address pair, address token, uint256 amountIn) external returns (uint256 liquidity);
+    // function removeLiquiditySingle(address pair, address token, uint256 lpTokenAmount) external returns (uint256 amount);
```

Comment:
I would just get rid of these given the safety concerns? Can always deploy a helper contract if people really want it. does Uniswap support it?

Reply:
neither UniV2 nor UniV3 supports adding liq for only one token. 

Technically it's not so hard... we just have to sell (around) half of the tokens into the other token right? 

You're still proposing to get rid of it?

Reply:
Yes, it's a dangerous and silly feature imo

Reply:
removed, fixed